core/src/main/scala/filodb.core/memstore/PartKeyTantivyIndex.scala

```diff
@@ -272,12 +398,134 @@ object ByteBufferEncodingUtils {
+  private final val TERM_IN_TYPE_BYTE: Byte = 4
+  private final val PREFIX_TYPE_BYTE: Byte = 5
+  private final val MATCH_ALL_TYPE_BYTE: Byte = 6
+  private final val LONG_RANGE_TYPE_BYTE: Byte = 7
+
+  private final val OCCUR_MUST: Byte = 1
+  private final val OCCUR_MUST_NOT: Byte = 2
+
+  private val buffer = {
+    val buffer = TantivyQueryBuilder.bufferLocal.get()
+    buffer.clear()
+
+    buffer
+  }
+
+  private def writeString(s: String): Unit = {
+    ByteBufferEncodingUtils.writeStringToBuffer(s, buffer)
+  }
+
+  private def writeLong(v: Long): Unit = {
+    buffer += v.toByte
+    buffer += (v >> 8).toByte
+    buffer += (v >> 16).toByte
+    buffer += (v >> 24).toByte
+    buffer += (v >> 32).toByte
+    buffer += (v >> 40).toByte
+    buffer += (v >> 48).toByte
+    buffer += (v >> 56).toByte
+  }
+
+  private def writeOccur(occur: PartKeyQueryOccur): Unit = {
+    occur match {
+      case OccurMust => buffer += OCCUR_MUST
+      case OccurMustNot => buffer += OCCUR_MUST_NOT
+    }
+  }
+
+  override protected def visitStartBooleanQuery(): Unit = {
+    if (buffer.nonEmpty) {
+      // Nested, add occur byte
+      buffer += OCCUR_MUST
+    }
+    buffer += BOOLEAN_TYPE_BYTE
+  }
+
+  override protected def visitEndBooleanQuery(): Unit = {
+    buffer += TERMINATOR_BYTE
+  }
+
+  override protected def visitEqualsQuery(column: String, term: String, occur: PartKeyQueryOccur): Unit = {
+    writeOccur(occur)
+
+    // Type byte, col len, col, term len, term
+    buffer += EQUALS_TYPE_BYTE
+    writeString(column)
+    writeString(term)
+  }
+
+  override protected def visitRegexQuery(column: String, pattern: String, occur: PartKeyQueryOccur): Unit = {
+    writeOccur(occur)
+
+    // Type byte, col len, col, pattern len, pattern
+    buffer += REGEX_TYPE_BYTE
+    writeString(column)
+    writeString(pattern)
+  }
+
+  override protected def visitTermInQuery(column: String, terms: Seq[String], occur: PartKeyQueryOccur): Unit = {
+    writeOccur(occur)
+
+    // Type byte, column len, column bytes, (for each term -> term len, term), 0 length
+    buffer += TERM_IN_TYPE_BYTE
+    writeString(column)
+
+    ByteBufferEncodingUtils.writeLengthToBuffer(terms.length, buffer)
+    for (term <- terms) {
+      writeString(term)
+    }
+  }
+
+  override protected def visitPrefixQuery(column: String, prefix: String, occur: PartKeyQueryOccur): Unit = {
+    writeOccur(occur)
+
+    // Type byte, col len, col, prefix len, prefix
+    buffer += PREFIX_TYPE_BYTE
+    writeString(column)
+    writeString(prefix)
+  }
+
+  override protected def visitMatchAllQuery(): Unit = {
+    buffer += OCCUR_MUST
+    buffer += MATCH_ALL_TYPE_BYTE
+  }
+
+  override protected def visitRangeQuery(column: String, start: Long, end: Long, occur: PartKeyQueryOccur): Unit = {
+    writeOccur(occur)
+
+    // Type byte, col len, col, start, end
+    buffer += LONG_RANGE_TYPE_BYTE
+    writeString(column)
+    writeLong(start)
+    writeLong(end)
+  }
```

Comment:
Where is this invoked? And should this have a unit test added, as well?

Reply:
This is invoked in the base class as part of visitQueryWithStartAndEnd.  Today, Tantivy won't get any calls to this as we made a change to send the start/end time as explicit parameters for optimization vs part of the query.  However, I think the right thing is to keep this implemented should we change that logic in the future.

You're correct that I'm missing a test for this case.  Will add one.

Reply:
Added a test